### PR TITLE
PHPStan (Level 0) on controllers/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 0) on controllers/
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #16471
| How to test?      | Github Actions is :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26518)

### CODE REVIEW

File: controllers/admin/AdminCartsController.php [hunk 1]
```diff
@@ -472,7 +472,7 @@ public function ajaxProcessUpdateQty()
             // Don't try to use a product if not instanciated before due to errors
             if (isset($product) && $product->id) {
                 if (($id_product_attribute = Tools::getValue('id_product_attribute')) != 0) {
-                    if (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !Attribute::checkAttributeQty((int) $id_product_attribute, (int) $qty)) {
+                    if (!Product::isAvailableWhenOutOfStock($product->out_of_stock) && !ProductAttribute::checkAttributeQty((int) $id_product_attribute, (int) $qty)) {
```
Comment (matthieu-rolland) [inline controllers/admin/AdminCartsController.php:475]: is this modification linked to phpstan requirements ?

Comment (Progi1984) [reply controllers/admin/AdminCartsController.php:475]: I think it's an omission when @atomiix has migrated to PHP 8.0.

**Review by matthieu-rolland — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Oh seems, there is a last error :) 
```
Internal error: Internal error: Undefined constant                         
     "_PS_DEFAULT_THEME_NAME_" in file                                          
     /home/runner/work/PrestaShop/PrestaShop/controllers/admin/AdminTranslatio  
     nsController.php
```

**Comment by Progi1984** [comment]
@PierreRambaud & @matthieu-rolland I fixed the error in PHP 8.0 / PHPStan with a modification in the `autoload.php`. Thanks.

**Review by PierreRambaud — DISMISSED** [review]

File: controllers/admin/AdminGendersController.php [hunk 2]
```diff
@@ -29,6 +29,15 @@
  */
 class AdminGendersControllerCore extends AdminController
 {
+    /**
+     * @var int
+     */
+    public $default_image_height;
+    /**
```
Comment (sowbiba) [inline controllers/admin/AdminGendersController.php:37]:
```suggestion

    /**
```

**Review by sowbiba — DISMISSED** [review]
Some YamlLint checks to fix. Otherwise seems good

Good job

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by Progi1984** [comment]
@boubkerbribri Could you launch the suite on this PR ?

File: controllers/admin/AdminGendersController.php (hunk 2 above)
Comment (Progi1984) [reply controllers/admin/AdminGendersController.php:37]: Fixed

**Comment by Progi1984** [comment]
@sowbiba @PierreRambaud @matthieu-rolland Just fixed an empty line.

File: controllers/admin/AdminImportController.php [hunk 3]
```diff
@@ -3733,10 +3734,6 @@ public function supplierImport($offset = false, $limit = false, $validateOnly =
 
             $info = AdminImportController::getMaskedRow($line);
 
-            if ($offset > 0) {
-                $this->toto = true;
```
Comment (matks) [inline controllers/admin/AdminImportController.php:3737]: 😆

**Review by matks — COMMENTED** [review]
@Progi1984 You are modifying some old and fragile code, not covered by tests, that do rely on a lot of magic. It is impossible to merge these code changes without a deep QA analysis and I think the only thing you can tell QA is "please test all the application, here is the list of controllers I modified".

I suggest to split the PR in 2 parts:
- changes that cannot modify the application behavior (like adding a new public property) and could be merged without long QA test
- changes that introduce a risk and require a long QA work

File: controllers/front/AttachmentController.php [hunk 4]
```diff
@@ -32,7 +32,7 @@ public function postProcess()
             Tools::redirect('index.php');
         }
 
-        Hook::exec('actionDownloadAttachment', ['attachment' => &$a]);
+        Hook::exec('actionDownloadAttachment', ['attachment' => $attachment]);
```
Comment (kpodemski) [inline controllers/front/AttachmentController.php]:
```suggestion
        Hook::exec('actionDownloadAttachment', ['attachment' => &$attachment]);
```

I think you could leave this as is to allow developers modify this var

**Review by kpodemski — COMMENTED** [review]
One little comment. I love these PRs 👍

File: controllers/front/AttachmentController.php (hunk 4 above)
Comment (Progi1984) [reply controllers/front/AttachmentController.php]: Fixed

**Comment by Progi1984** [comment]
@matks For each PHPStan PR, @boubkerbribri executes a suite of tests in automated way for having some errors. Actually, these have detected twice bugs on all PHPStan PRs.

**Comment by matks** [comment]
> @matks For each PHPStan PR, @boubkerbribri executes a suite of tests in automated way for having some errors. Actually, these have detected twice bugs on all PHPStan PRs.

It's not enough for me.

Look at this bug report: https://github.com/PrestaShop/PrestaShop/issues/26606

The original issue was a dynamic property that was added to the class. It's a good change, but it had unexpected consequences.

Controllers are the main interface for the application. It's maybe the 2nd most critical piece of code of PrestaShop legacy part after the ObjectModels. I think we cannot rely only the test suite here.

**Comment by Progi1984** [comment]
@matks No problem : I will ask some exploratory tests to QA Team

**Review by atomiix — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
Hey @boubkerbribri , if you're able to run the full campaign on it? :)

**Review by boubkerbribri — DISMISSED** [review]

**Comment by boubkerbribri** [comment]
@Progi1984 Can you resolve the conflict please ?

**Comment by Progi1984** [comment]
@boubkerbribri Rebase done (Ping @atomiix & @PierreRambaud)

**Review by atomiix — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Comment by nesrineabdmouleh** [comment]
All good for UI tests
thanks @Progi1984

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh

**Comment by matks** [comment]
Hi folks just to reassure me, we did some exploratory tests on this PR ? 😄

**Comment by Progi1984** [comment]
Oh #fail. I have misread @nesrineabdmouleh. Sorry.

**Pull request merged**